.github/workflows/sync-main-to-prod.yml

```diff
@@ -66,6 +76,19 @@ jobs:
               fi;
           }
 
+          # checks whether a filename matches timestamp.json, snapshot.json, or [0-9]+.snapshot.json. If not,
+          # this workflow will exit as we only want to run it when ONLY these files are changed.
+          check_filename() {
+            #TODO 
+          }
+
+          # Diff main and prod to determine whether ONLY the timestamp and snapshot files have changed in main.
+          # If other files have also changed, exit - in this case, the sync should be to preprod.
+          # Note: other non-timestamp/snapshot changes should only occur during a ceremony, and
+          # will go through the sync-ceremony-* flow that hits main and preprod. This means there should never 
+          # be changed files in prod that have not also hit the main branch.
+          diff -qr repository/repository sigstore-tuf-root | grep -Po '([0-9\.]*\w+\.\w+(?= differ))|((Only in \w+\: )\K(.*))' | while read l; do check_filename $l; done
```

Comment:
Will need to download the bucket metadata first before this command runs.

Reply:
Correct!

Reply:
Thanks, it makes sense now that I've looked at it again.